### PR TITLE
store/tikv: return error when transaction is too large.

### PR DESCRIPTION
**Pull request by coocood**
The maximun transaction lock time is 120s, when transaction is larger than 400M, we returns error
to prevent the large transaction harm the entire cluster.

As test for this is too expensive, I just tested it manually.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @disksing PTAL

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func newTwoPhaseCommitter(txn *tikvTxn) (*twoPhaseCommitter, error) {
 			lockTTL = defaultLockTTL
 		}
 		if lockTTL > maxLockTTL {
-			lockTTL = maxLockTTL
+			return nil, kv.ErrTxnTooLarge
```
Comment (disksing) [inline store/tikv/2pc.go]: Please update comment at L113.

**Comment by ngaut** [comment]
400MiB is huge, how about 50M

**Comment by shenli** [comment]
50MB is too small, especially when we use load data command to load data.

**Comment by coocood** [comment]
@ngaut 
Since old version doesn't have any limit, change it to a small value may cause old cluster stop working.

At least our system can afford 400MB transaction with longer lock time.

We can't set it to 400MB first, change it to smaller value later if we find large transaction is problematic.

**Comment by shenli** [comment]
@ngaut What's your opinion？

**Comment by ngaut** [comment]
LGTM， let's make it smaller step by step.

**Comment by ngaut** [comment]
I am still worrying about if users delete millions of row, it could be trouble.

**Comment by shenli** [comment]
@ngaut I think we should test deleting large scale data in RC2.

**Comment by ngaut** [comment]
Please add an issue for testing large delete transaction.

**Comment by gejigeji** [comment]
I also meet the problem

mysql> delete from t_table where 0=0;
ERROR 1105 (HY000): transaction is too large

**Comment by coocood** [comment]
@gejigeji 
Delete a table without a condition is considered an expensive query.
If you really want to delete all rows, you can use `truncate table t_table`.

**Comment by gejigeji** [comment]
@coocood thks, I got

**Pull request merged**